File: demon/gearman_worker.py
import os
import gearman

#create a worker
worker = gearman.GearmanWorker(["127.0.0.1"])

def do_judge(code_no):
	#print os.system( "./python judge -n%s" % code_no )       
	print "I`s work!"

worker.register_task( "do_judge" , do_judge )
print worker.work()

File: demon/gearman_worker.php
<?php
$worker= new GearmanWorker(); 
$worker->addServer("172.17.0.46",4730);
 
$worker->addFunction("do_judge", "do_judge");
   
	while (1){
      	     
   		print "Waiting for job...\n";
	     	$ret= $worker->work();
		if ($worker->returnCode() != GEARMAN_SUCCESS)
			break;
	}

function do_judge($job){

	$workload= $job->workload();
	var_dump( $workload );
	echo "Received job: " . $job->handle() . "\n";
	echo "Workload: $workload\n";
        @$result=system("python judge.py -n$workload" );

	//将测试结果写入数据库中	
	return $result;
}
?>

